src/Hooks.php

```diff
@@ -2,93 +2,129 @@
-		static $instance = null;
-		if ( $instance == null ) {
-			global $wgHooks;
-			$instance = new Hooks;
-			$wgHooks['InternalParseBeforeSanitize'][] = $instance;
-			$wgHooks['ParserBeforeInternalParse'][] = $instance;
-		}
+	public function onParserFirstCallInit( $parser ): void {
 		$parser->setFunctionHook(
 			'defaultlink',
-			[ $instance, 'linkParserFunction' ],
+			[ $this, 'linkParserFunction' ],
 			SFH_NO_HASH | SFH_OBJECT_ARGS
 		);
-		$parser->setHook( 'nodefaultlinks', [ $instance, 'noLinksTag' ] );
-
-		return true;
-	}
-
-	/* Delete stored properties for a given page ID.
-	 */
-	private static function deleteProps( $page_id ) {
-		wfGetDB( DB_PRIMARY )->delete(
-			'page_props',
-			[ 'pp_page' => $page_id, 'pp_propname' => [ 'defaultlink', 'defaultlinksec' ] ],
-			__METHOD__
-		);
+		$parser->setHook( 'nodefaultlinks', [ $this, 'noLinksTag' ] );
 	}
 
-	/* Returns whether the article is in a namespace that is allowed to define
+	/**
+	 * Returns whether the article is in a namespace that is allowed to define
 	 * incoming link formatting.
+	 * @param PageReference $title
+	 * @return bool
 	 */
-	private static function nsHasFormattedLinks( LinkTarget $title ) {
-		global $wgDFEnabledNamespaces;
+	private function nsHasFormattedLinks( PageReference $title ) {
+		// It's not possible to inject configuration overrides into parser tests in time for
+		// article fixture initialization, so hardcode the allowed set of namespaces there.
+		if ( defined( 'MW_PARSER_TEST' ) ) {
+			return $title->getNamespace() === NS_MAIN;
+		}
 
-		return in_array( $title->getNamespace(), $wgDFEnabledNamespaces );
+		return in_array( $title->getNamespace(), $this->config->get( 'DFEnabledNamespaces' ) );
 	}
 
-	/* Serialize the defaultlinksec property before updating it in database.
+	/**
+	 * Serialize the defaultlinksec property before updating it in the database.
+	 * @param Title $title
+	 * @param RenderedRevision $renderedRevision
+	 * @param DeferrableUpdate[] &$updates
 	 */
-	public static function onLinksUpdateConstructed( LinksUpdate &$lu ) {
-		if ( isset( $lu->mProperties ) && isset( $lu->mProperties['defaultlinksec'] ) ) {
+	public function onRevisionDataUpdates( $title, $renderedRevision, &$updates ): void {
+		$parserOutput = $renderedRevision->getRevisionParserOutput();
+		$defaultLinkSec = $parserOutput->getExtensionData( 'defaultlinksec' );
+		if ( $defaultLinkSec !== null ) {
 			$s = '';
-			foreach ( $lu->mProperties['defaultlinksec'] as $k => $v ) {
+			foreach ( $defaultLinkSec as $k => $v ) {
 				$s .= ( $s == '' ? '' : "\n" ) . $k . "\n" . $v;
 			}
-			$lu->mProperties['defaultlinksec'] = $s;
+			$parserOutput->setPageProperty( 'defaultlinksec', $s );
+			$parserOutput->setExtensionData( 'defaultlinksec', null );
 		}
-
-		return true;
-	}
-
-	/* Delete link properties when deleting an article.
-	 */
-	public static function onPageDeleteComplete( &$article, &$user, $reason, $id ) {
```

Comment:
Is there a replacement for that hook?

Reply:
It's not needed—`LinksDeletionUpdate` takes care of pruning `page_props` when a page is deleted.